aesara/tensor/blockwise.py

```diff
@@ -0,0 +1,315 @@
+
+    def __init__(self, op, signature=None):
+        self.op = op
+        self.signature = signature or self.op.gufunc_sig
+
+    def get_output_info(self, *inputs):
+        """Return the outputs dtype and broadcastable pattern and the
+        dimshuffled inputs.
+
+        """
+        target_length = max(input.type.ndim for input in inputs)
+        args = []
+        for input in inputs:
+            length = input.type.ndim
+            difference = target_length - length
+            if not difference:
+                args.append(input)
+            else:
+                # TODO: use LComplete instead
+                args.append(
+                    DimShuffle(
+                        input.type.broadcastable,
+                        ["x"] * difference + list(range(length)),
+                    )(input)
+                )
+        inputs = args
+
+        # TODO: Correct this
+        out_dtype = inputs[0].dtype
+
+        bcast_shape, dim_sizes = _parse_input_dimensions(inputs, self.signature[0])
+        output_shapes = _calculate_shapes(bcast_shape, dim_sizes, self.signature[1])
+
+        return out_dtype, output_shapes, inputs
+
+    def make_node(self, *inputs):
+        num_expected_inps = len(self.signature[0])
+        if len(inputs) != num_expected_inps:
+            raise ValueError(
+                f"Expected {int(num_expected_inps)} inputs, got {len(inputs)}"
+            )
+
+        out_dtype, output_shapes, inputs = self.get_output_info(*inputs)
+
+        def safe_const_val(x):
+            try:
+                return get_scalar_constant_value(x)
+            except NotScalarConstantError:
+                return None
+
+        outputs = [
+            TensorType(out_dtype, shape=tuple(safe_const_val(s) for s in shp))()
+            for shp in output_shapes
+        ]
+        return Apply(self, list(inputs), outputs)
+
+    def infer_shape(self, fgraph, node, shapes):
+        bcast_shape, dim_sizes = _parse_input_dimensions(node.inputs, self.signature[0])
+        output_shapes = _calculate_shapes(bcast_shape, dim_sizes, self.signature[1])
+        return output_shapes
+
+    def L_op(self, inputs, outs, ograds):
+        # Compute grad with respect to broadcasted input
+        rval = self._bgrad(inputs, outs, ograds)
+
+        # sum out the broadcasted dimensions
+        for i, ipt in enumerate(inputs):
+            if isinstance(rval[i].type, (NullType, DisconnectedType)):
+                continue
+
+            # List of all the dimensions that are broadcastable for input[i] so
+            # we can sum over them
+            # TODO: only count dimensions that were effectively broadcasted
+            to_sum = [
+                j
+                for j, bcast in enumerate(ipt.type.broadcastable)
+                if bcast and not outs[0].broadcastable[j]
+            ]
+
+            if to_sum:
+                sr = at_sum(rval[i], axis=to_sum, keepdims=True)
+                rval[i] = sr
```

Comment:
We should probably address this `Elemwise`-originating approach, because it fails now that we've extended Aesara's broadcasting capabilities (see https://github.com/aesara-devs/aesara/issues/1089).  [This is a viable approach](https://github.com/aesara-devs/aesara/issues/1089#issuecomment-1205615007) we can use in the interim&mdash;after swapping those tensor-typed `Op`s with their scalar equivalents, though.